apollo-router/src/plugin/mod.rs

```diff
@@ -69,8 +68,10 @@ pub struct PluginInit<T> {
     /// Router Supergraph Schema (schema definition language)
     pub supergraph_sdl: Arc<String>,
     /// The supergraph schema (parsed)
-    pub(crate) supergraph_schema:
-        Arc<apollo_compiler::validation::Valid<apollo_compiler::schema::Schema>>,
+    pub(crate) supergraph_schema: Arc<Valid<Schema>>,
+
+    /// The parsed subgraph schemas from the query planner, keyed by subgraph name
+    pub(crate) subgraph_schemas: Arc<HashMap<String, Valid<Schema>>>,
```

Comment:
```suggestion
    pub(crate) subgraph_schemas: Arc<HashMap<String, Arc<Valid<Schema>>>>,
```
if a subgraph plugin wants to keep a copy of the schema around, then we should avoid copying the entire structure

Reply:
Thanks for the feedback. Updated to use Arc